app/scripts/views/sign_up.js

```diff
@@ -258,6 +258,12 @@ define(function (require, exports, module) {
         return true;
       }
 
+      // some users input a "@firefox" as the email field.
+      // this will never be a valid email, therefore we block the attempt.
+      if (String.prototype.hasOwnProperty('endsWith') && email.endsWith('@firefox')) {
```

Comment:
I may be over-thinking this but do we have an approach in place for polyfilling ES6 methods on older user agents anywhere? It would be nicer if the special case code was for the old clients and tucked away in a dedicated module, then we could happily use `endsWith` and its ilk everywhere without writing .